### PR TITLE
[MINDEXER-132] use Java 7 Files.createTempDirectory() API

### PR DESCRIPTION
**Pull request by hboutemy**
https://issues.apache.org/jira/browse/MINDEXER-132

### CODE REVIEW

File: indexer-core/src/main/java/org/apache/maven/index/updater/DefaultIndexUpdater.java [hunk 1]
```diff
@@ -190,9 +191,7 @@ private Date loadIndexDirectory( final IndexUpdateRequest updateRequest, final R
         {
             updateRequest.getIndexTempDir().mkdirs();
         }
-        File indexDir = File.createTempFile( remoteIndexFile, ".dir" , updateRequest.getIndexTempDir() );
-        indexDir.delete();
-        indexDir.mkdirs();
+        File indexDir = Files.createTempDirectory( updateRequest.getIndexTempDir().toPath(), remoteIndexFile ).toFile();
```
Comment (cstamas) [inline indexer-core/src/main/java/org/apache/maven/index/updater/DefaultIndexUpdater.java:194]: updateRequest.getIndexTempDir() is nullable, above is nullcheck performed. Better make `File indexDir` above IF and then update it in IF (add ELSE as well).

File: indexer-reader/src/test/java/org/apache/maven/index/reader/TestSupport.java [hunk 2]
```diff
@@ -54,9 +56,7 @@
     public void setup()
         throws IOException
     {
-        this.tempDir = new File( "target/tmp-" + getClass().getSimpleName() );
-        this.tempDir.delete();
-        this.tempDir.mkdirs();
+        this.tempDir = Files.createTempDirectory( getClass().getSimpleName() + ".temp" ).toFile();
         this.directoryResourceHandlers = new ArrayList<>();
```
Comment (gnodet) [inline indexer-reader/src/test/java/org/apache/maven/index/reader/TestSupport.java:60]: I don't like creating temp directories in tests:
  * it's not deleted
  * it's hard to find
  * if the directory is deleted at the end of the test, its not available anymore to debug the test if needed

**Review by gnodet — CHANGES_REQUESTED** [review]

File: indexer-reader/src/test/java/org/apache/maven/index/reader/TestSupport.java (hunk 2 above)
Comment (cstamas) [reply indexer-reader/src/test/java/org/apache/maven/index/reader/TestSupport.java:60]: We usually when needed to create temp dirs, use property for surefilre/failsafe to set tmp dir on CI to some place that is nuked along with workspace....

Comment (gnodet) [reply indexer-reader/src/test/java/org/apache/maven/index/reader/TestSupport.java:60]: I see https://github.com/apache/maven-indexer/blob/master/indexer-core/pom.xml#L179 which looks fine.

**Review by gnodet — APPROVED** [review]

**Review by cstamas — APPROVED** [review]

**Pull request merged**